utils/src/server/rpc_module.rs

```diff
@@ -35,21 +37,44 @@ pub type MethodSink = mpsc::UnboundedSender<String>;
 
 type Subscribers = Arc<Mutex<FxHashMap<(ConnectionId, SubscriptionId), (MethodSink, oneshot::Receiver<()>)>>>;
 
-/// Identifier of the method type.
-#[derive(Debug, Copy, Clone)]
-pub enum MethodType {
+/// Callback wrapper that can be either sync or async.
+pub enum MethodCallback {
 	/// Synchronous method handler.
-	Sync,
+	Sync(SyncMethod),
 	/// Asynchronous method handler.
-	Async,
+	Async(AsyncMethod),
+}
+
+impl MethodCallback {
+	/// Execute the callback, sending the resulting JSON to the specified sink.
+	pub async fn execute(&self, tx: &MethodSink, req: JsonRpcRequest<'_>, conn_id: ConnectionId) {
+		let id = req.id.clone();
+
+		let result = match self {
+			MethodCallback::Sync(callback) => {
+				let params = RpcParams::new(req.params.map(|params| params.get()));
```

Comment:
You could hoist this outside the match and then do `let params = OwnedRpcParams::from(params);` on line 60.